packages/editor/src/components/block-list/block.js

```diff
@@ -394,7 +393,7 @@ export class BlockListBlock extends Component {
 		// We render block movers and block settings to keep them tabbale even if hidden
 		const shouldRenderMovers = ( isSelected || hoverArea === 'left' ) && ! showEmptyBlockSideInserter && ! isMultiSelecting && ! isPartOfMultiSelection && ! isTypingWithinBlock;
 		const shouldRenderBlockSettings = ( isSelected || hoverArea === 'right' ) && ! isMultiSelecting && ! isPartOfMultiSelection;
-		const shouldShowBreadcrumb = isHovered && ! isEmptyDefaultBlock;
+		const shouldShowBreadcrumb = ( isSelected || isHovered ) && ! isEmptyDefaultBlock;
```

Comment:
This makes the hover label to stick when the block receives a click, and making it visible when a block is being edited:

![hover-on-block-edit](https://user-images.githubusercontent.com/583546/44340459-35803380-a485-11e8-9a32-d37c7a378dc9.png)

Reply:
I think this is a bad idea. **Nothing should _ever_ constantly overlap the content of the selected block.** It is bad enough that sibling inserters currently do this; see #8206, #8881, and #8883.

If the hover label was shown outside of the block border, then I think this would be fine. Of course, that would conflict with the toolbar. I think that dragging a selected block should be done from empty space in the toolbar (or just dragging from anywhere in the toolbar, similar to what GNOME does), and/or dragging from the up/down movers on the side.

See also:
- #6224
- #8880
- #9074
- #9075
- #9075

Reply:
Just want to note I'm also pretty uncomfortable with the handle being visible when the block is selected. Feels like additional weight in the UI.